client/quantum_serverless/core/events.py

```diff
@@ -0,0 +1,136 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+================================================
+Provider (:mod:`quantum_serverless.core.events`)
+================================================
+
+.. currentmodule:: quantum_serverless.core.events
+
+Quantum serverless events handler
+=================================
+
+.. autosummary::
+    :toctree: ../stubs/
+
+    ExecutionMessage
+    EventHandler
+    RedisEventHandler
+"""
+
+import json
+import os
+import time
+from abc import ABC
+from dataclasses import dataclass, asdict
+from typing import Optional, Dict, Any
+
+import redis
+
+from quantum_serverless.core.constrants import (
+    QS_EVENTS_REDIS_PASSWORD,
+    QS_EVENTS_REDIS_PORT,
+    QS_EVENTS_REDIS_HOST,
+)
+
+
+@dataclass
+class ExecutionMessage:
+    """ExecutionMessage class for meta information about function execution."""
+
+    workload_id: str
+    uid: str
+    layer: str
+    resources: Dict[str, float]
+    function_meta: Dict[str, Any]
+    timestamp: float = time.time()
+
+    def to_dict(self):
+        """Converts to dictionary."""
+        return asdict(self)
+
+    def __repr__(self):
+        return f"<ExecutionMessage | {self.workload_id}|{self.layer}>"
+
+
+class EventHandler(ABC):  # pylint: disable=too-few-public-methods
```

Comment:
Can we implement some subscribe interface in case that someone wants to listen some specific events? (or some events by some regexp)